cmd/execution_builder.go

```diff
@@ -422,7 +433,7 @@ func (exeNode *ExecutionNode) LoadProviderEngine(
 		node.Logger,
 		providerMetrics,
 		execution_data.DefaultSerializer,
-		bs,
+		exeNode.blobService,
```

Comment:
Let's add a check to ensure this module has been set and is not `nil`. Type system can't ensure this, so we'd better add explicit check.